### PR TITLE
Refactor the layout so src files are organised and tests are separated

### PR DESCRIPTION
**Pull request by bosakad**
finished layout + new readme (sorry did not see the new branch) + black + pytest + updated docs

### CODE REVIEW

**Comment by mishagli** [comment]
I have moved the changes related to README.md to its branch, as it belongs to the Issue #2.

**Review by mishagli — CHANGES_REQUESTED** [review]
This notebook shouldn't be here. It rather belongs to the `/examples`.

**Review by mishagli — CHANGES_REQUESTED** [review]
About `/src/other/`

You are correct that you want to move some kind of utility functions to a separate folder. But it is a good practice to keep meaningful names for these as well. I'd suggest to name this folder `/utility_tools` or something like that.

The name `/other` is quite bad for several reasons. Firstly, it doesn't follow from its name what's inside. Secondly, this makes one wanting to dive into that folder because of non-needed-here curiosity :)

**Review by mishagli — CHANGES_REQUESTED** [review]
This file `src/timing/unaryTesting.py` shouldn't be here. What is it testing? If it is a test for your module, it should be in `/tests`. If this is a script where you played around, it should go to `/examples`.

File: setup.py [hunk 1]
```diff
@@ -8,5 +21,9 @@
 	author='Bosak Adam',
 	author_email='bosadam@seznam.cz',
 	url='https://github.com/bosakad/GP-Optimization',
-	packages=find_packages()
+	packages=find_packages(),
+	cmdclass={
+        'clean': CleanCommand,
+    }
+
```
Comment (mishagli) [inline setup.py:28]: I would suggest exclude those directories, where you have no packages. You packages are in `/src`. So you don't want that whatever is in `/output` and `/examples` will be written to a PATH. Particularly, this `/output` directory may have lots of different data that will probably appear there from run to run.

To do this, you should write:
```python
packages=setuptools.find_namespace_packages(exclude=["tests", "docs","examples","inputs_outputs"]),
```

**Review by mishagli — CHANGES_REQUESTED** [review]

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	description='GP optimization project',
+	author='Bosak Adam',
+	author_email='bosadam@seznam.cz',
+	url='https://github.com/bosakad/GP-Optimization',
+	packages=find_packages(),
+	cmdclass={
+        'clean': CleanCommand,
+    }
+
+)
```
Comment (mishagli) [inline setup.py:29]: Please add the requirements from `requirements.txt` to `setup.py`. They should go to

```python
install_requires=[
        "cvxpy>=1.2.1",
        "cycler>=0.11.0",
        "numpy>=1.22", # and so on, please write all the packages you need here
    ],
```

Please note that I write `>=`. It is generally better to have a flexibility in packages rather that require very paricular version. You never know what changes those guys in Numy or Mosek may do so a particular version becomes deprecated or so. So we should keep upper bound open :)

**Review by mishagli — CHANGES_REQUESTED** [review]

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from setuptools import find_packages
+import os
+from setuptools import setup, Command
```
Comment (mishagli) [inline setup.py:3]: Too clumpy imports. You are importing from `setuptools` in two different lines. Surely, it should be done in one line.

So your first two lines (note the order) should be:

```python
import os
from setuptools import setup, find_packages, Command
```

**Review by mishagli — CHANGES_REQUESTED** [review]
Change import to be
```python
import os
from setuptools import setup, find_packages, Command
```

**Comment by bosakad** [comment]
> Change import to be
> 
> ```python
> import os
> from setuptools import setup, find_packages, Command
> ```

all done and should be working:)

**Comment by bosakad** [comment]
> Change import to be
> 
> ```python
> import os
> from setuptools import setup, find_packages, Command
> ```

**Pull request merged**